src/Open3D/Utility/FileSystem.cpp

```diff
@@ -218,6 +231,105 @@ FILE *FOpen(const std::string &filename, const std::string &mode) {
     return fp;
 }
 
+static std::string GetIOErrorString(const int errnoVal) {
+    switch (errnoVal) {
+        case EPERM:
+            return "Operation not permitted";
+        case EACCES:
+            return "Access denied";
+        // Error below could be EWOULDBLOCK on Linux
+        case EAGAIN:
+            return "Resource unavailable, try again";
+#if !defined(WIN32)
+        case EDQUOT:
+            return "Over quota";
+#endif
+        case EEXIST:
+            return "File already exists";
+        case EFAULT:
+            return "Bad filename pointer";
+        case EINTR:
+            return "open() interrupted by a signal";
+        case EIO:
+            return "I/O error";
+        case ELOOP:
+            return "Too many symlinks, could be a loop";
+        case EMFILE:
+            return "Process is out of file descriptors";
+        case ENAMETOOLONG:
+            return "Filename is too long";
+        case ENFILE:
+            return "File system table is full";
+        case ENOENT:
+            return "No such file or directory";
+        case ENOSPC:
+            return "No space available to create file";
+        case ENOTDIR:
+            return "Bad path";
+        case EOVERFLOW:
+            return "File is too big";
+        case EROFS:
+            return "Can't modify file on read-only filesystem";
+#if EWOULDBLOCK != EAGAIN
+        case EWOULDBLOCK:
+            return "Operation would block calling process";
+#endif
+        default: {
+            std::stringstream s;
+            s << "IO error " << errnoVal << " (see sys/errno.h)";
+            return s.str();
+        }
+    }
+}
+
+bool FReadToBuffer(const std::string &path,
+                   std::vector<char> &bytes,
+                   std::string *errorStr) {
+    bytes.clear();
+    if (errorStr) {
+        errorStr->clear();
+    }
+
+    FILE *file = FOpen(path.c_str(), "rb");
+    if (!file) {
+        if (errorStr) {
+            *errorStr = GetIOErrorString(errno);
+        }
+
+        return false;
+    }
+
+    if (fseek(file, 0, SEEK_END) != 0) {
+        // We ignore that fseek will block our process
+        if (errno && errno != EWOULDBLOCK) {
+            if (errorStr) {
+                *errorStr = GetIOErrorString(errno);
+            }
+
+            fclose(file);
+            return false;
+        }
+    }
+
+    const size_t filesize = ftell(file);
+    rewind(file);  // reset file pointer back to beginning
+
+    bytes.resize(filesize);
```

Comment:
Historical reasons. It was originally copied from Filament, where it is simpler and uses read(2) rather than fread(3), if I recall correctly. I think it's also duplicated somewhere, but I don't remember where. I know I unduplicated it at one point, and that got undone somehow, maybe in a merge conflict or something.